src/FloatingDockContainer.cpp

```diff
@@ -560,19 +562,24 @@ void CFloatingDockContainer::updateWindowTitle()
 	auto TopLevelDockArea = d->DockContainer->topLevelDockArea();
 	if (TopLevelDockArea)
 	{
-		d->setWindowTitle(TopLevelDockArea->currentDockWidget()->windowTitle());
+		CDockWidget* CurrentWidget = TopLevelDockArea->currentDockWidget();
+		d->setWindowTitle(CurrentWidget->windowTitle());
+		setWindowIcon(CurrentWidget->icon());

```

Comment:
Set window icon only if the dock widget icon returns a valid icon

Reply:
Fixed